src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -89,53 +89,112 @@
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(5);           // OK
+ *   }
+ *
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
+ *   }
+ * }
+ * </pre>
  * <p>
- * To configure to check the {@code this} qualifier for fields only:
+ * To configure the check for fields only:
  * </p>
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;&gt;
  *   &lt;property name=&quot;checkMethods&quot; value=&quot;false&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
- * <p>
- * Examples of how the check works if validateOnlyOverlapping option is set to true:
- * </p>
+ * <p>Example:</p>
  * <pre>
- * public static class A {
- *   private int field1;
- *   private int field2;
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
  *
- *   public A(int field1) {
- *     // Overlapping by constructor argument.
- *     field1 = field1; // violation: Reference to instance variable "field1" needs "this".
- *     field2 = 0;
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(5);           // OK, no validation for methods
  *   }
  *
- *   void foo3() {
- *     String field1 = "values";
- *     // Overlapping by local variable.
- *     field1 = field1; // violation:  Reference to instance variable "field1" needs "this".
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
  *   }
  * }
+ * </pre>
+ * <p>
+ * To configure the check for methods only:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;RequireThis&quot;&gt;
+ *   &lt;property name=&quot;checkFields&quot; value=&quot;false&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
```

Comment:
there is no example of violation in this example. Is there a reason for it?

Reply:
Yes, that is because there is no such case where a method would be overlapped, so `this` is not required, and no violation is raised. But this is different from the previous example, as the violation for instance variable "c" disappears.